src/partition.rs

```diff
@@ -21,15 +21,60 @@ use crate::plane::*;
 use crate::predict::*;
 use crate::util::*;
 
-pub const NONE_FRAME: usize = 8;
-pub const INTRA_FRAME: usize = 0;
-pub const LAST_FRAME: usize = 1;
-pub const LAST2_FRAME: usize = 2;
-pub const LAST3_FRAME: usize = 3;
-pub const GOLDEN_FRAME: usize = 4;
-pub const BWDREF_FRAME: usize = 5;
-pub const ALTREF2_FRAME: usize = 6;
-pub const ALTREF_FRAME: usize = 7;
+// LAST_FRAME through ALTREF_FRAME correspond to slots 0-6.
+#[derive(PartialEq, Copy, Clone)]
+pub enum RefType {
+  NONE_FRAME = 8,
+  INTRA_FRAME = 0,
+  LAST_FRAME = 1,
+  LAST2_FRAME = 2,
+  LAST3_FRAME = 3,
+  GOLDEN_FRAME = 4,
+  BWDREF_FRAME = 5,
+  ALTREF2_FRAME = 6,
+  ALTREF_FRAME = 7,
+}
+
+impl RefType {
+  // convert to a ref list index, 0-6 (INTER_REFS_PER_FRAME)
+  pub fn to_index(self) -> usize {
+    match self {
+      NONE_FRAME => { panic!("Tried to get slot of NONE_FRAME"); },
+      INTRA_FRAME => { panic!("Tried to get slot of INTRA_FRAME"); },
+      _ => { return (self as usize) - 1; }
+    }
+  }
+  pub fn is_fwd_ref(self) -> bool {
+    (self as usize) < 5
+  }
+  pub fn is_bwd_ref(self) -> bool {
+    (self as usize) >= 5
+  }
+}
+
+use RefType::*;
+
+pub const ALL_INTER_REFS: [RefType; 7] = [
+  LAST_FRAME,
+  LAST2_FRAME,
+  LAST3_FRAME,
+  GOLDEN_FRAME,
+  BWDREF_FRAME,
+  ALTREF2_FRAME,
+  ALTREF_FRAME
+];
+
+pub const ALL_REFS: [RefType; 9] = [
```

Comment:
Seems like there's some risk we could confuse the index used for ALL_INTER_REFS and the one for ALL_REFS, maybe they should be enums too ?

Reply:
How about I move ALL_REFS to inside get_mvref_ref_frames, the only place it's used? (I actually want to delete that function entirely if possible...)

Reply:
sure, especially If it's going to be removed eventually then no need to make an enum for it.